doc/man/man5/strxnet.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH STRXNET 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
OpenSS7
.I STREAMS
add-on packages are separate add-on packages to the
.B "Linux Fast-STREAMS"
.[
lfs
.]
release that provide additional functionality and protocol support not traditionally found on UNIX
systems, or which have tranditionally fallen into the domain of valud-add system software, or, in
this case, where functionality was optional or differently handled in a number of implementations.
The
.B @PACKAGE_TITLE@
package is one of the several optional add-on packages that provide enhanced network protocol
support beyond that provided by the base
.I STREAMS
package.  This package provides support for communications protocols based on the
.IR "Transport Provider Interface (TPI)" ,
.BR tpi (7),
not provided by the base package.
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.I STREAMS
modules and drivers, libraries, programs, init scripts, and daemons that implement networking
protocols based on the transport layer of the OSI reference model.
.PP
The package builds and installs the following
.I STREAMS
modules (as kernel modules):
.IP \(en \w'\(en\(em'u
.B streams_timod.ko
module
.RI \(lq timod \(rq.
.PD 0
.IP \(en
.B streams_tirdwr.ko
module
.RI \(lq tirdwr \(rq.
.PD
.PP
The following are
.I STREAMS
modules installed by the
.B @PACKAGE@
package:
.TP
.BR tirdwr (4)
The
.BR tirdwr (4)
module is a module that can be pushed over a Stream conforming to the
Transport Provider Interface (TPI),
.[
tpi revision 2.0.0
.]
.BR tpi (7),
that provides a
.BR read (2s)
.BR write (2s)
interface to the transport provider.
See
.BR tirdwr (4)
for more information.
.TP
.BR timod (4)
The
.BR timod (4)
module is a module that cooperates with the
.BR xti (3)
library and is pushed by the library over a Stream conforming to the
Transport Provider Interface (TPI),
.[
tpi revision 2.0.0
.]
.BR tpi (7),
that provides XTI/TLI library access to the trasnport provider.
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2 STREAMS
driver, modules, libraries, utilities, test programs, daemons, and development environment for the
development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.I SVR 4.2 STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package is complete and has been in production use for
.I Linux Fast-STREAMS
for a number of years.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package contains the following manual pages:
.\"
.HP
.BR strxnet (5)\ \- \ 
@PACKAGE_TITLE@ package.
.PD 0
.HP
.BR strxnet_mknod (8)\ \- \ 
create or remove
.I STREAMS
XNET device nodes.
.PD
.\"
.SS "XTI Modules"
.PP
.HP
.BR timod (4)\ \- \ 
a
.I STREAMS
XTI/TLI compatibility interface.
.PD 0
.HP
.BR tirdwr (4)\ \- \ 
a
.I STREAMS
XTI/TLI read write compatiblity module.
.PD
.\"
.SS "XTI/TLI Library"
.PP
.HP
.BR libxnet (3)\ \- \ 
XNS 5.2 XTI/TLI Library (libxnet).
.PD 0
.HP
.BR xnet (3)\ \- \ 
XNS 5.2 XTI/TLI Library (libxnet).
.HP
.BR xti (3)\ \- \ 
XNS 5.2 XTI/TLI Library (libxnet).
.HP
.BR netbuf (3)\ \- \ 
the XTI/TLI netbuf structure.
.HP
.BR t_accept (3)\ \- \ 
accept a connection request.
.HP
.BR t_addleaf (3)\ \- \ 
add a point to a point-to-multipoint connection.
.HP
.BR t_alloc (3)\ \- \ 
allocate a library structure.
.HP
.BR t_bind (3)\ \- \ 
bind to a transport endpoint.
.HP
.BR t_close (3)\ \- \ 
close a transport endpoint.
.HP
.BR t_connect (3)\ \- \ 
establish a transport connection.
.HP
.BR t_errlist (3)\ \- \ 
describe an XTI/TLI library error.
.HP
.BR t_errno (3)\ \- \ 
TLI error number.
.HP
.BR t_error (3)\ \- \ 
print an error message.
.HP
.BR t_errstr (3)\ \- \ 
describe an XTI/TLI library error.
.HP
.BR t_free (3)\ \- \ 
free an XTI/TLI library structure.
.HP
.BR t_getinfo (3)\ \- \ 
get protocol information.
.HP
.BR t_getprotaddr (3)\ \- \ 
get protocol addresses.
.HP
.BR t_getstate (3)\ \- \ 
get interface state.
.HP
.BR t_listen (3)\ \- \ 
listen for a transport connection request.
.HP
.BR t_look (3)\ \- \ 
look for a transport event.
.HP
.BR t_nerr (3)\ \- \ 
describe an XTI/TLI library error.
.HP
.BR t_open (3)\ \- \ 
open a transport endpoint.
.HP
.BR t_optmgmt (3)\ \- \ 
manage transport options.
.HP
.BR t_rcv (3)\ \- \ 
receive transport data.
.HP
.BR t_rcvconnect (3)\ \- \ 
receive connection request confirmation.
.HP
.BR t_rcvdis (3)\ \- \ 
receive abortive disconnect.
.HP
.BR t_rcvleafchange (3)\ \- \ 
receive addition of point to point-to-multipoint connection.
.HP
.BR t_rcvopt (3)\ \- \ 
receive unit data.
.HP
.BR t_rcvrel (3)\ \- \ 
acknowledge orderly release request.
.HP
.BR t_rcvreldata (3)\ \- \ 
acknowledge orderly release request with data.
.HP
.BR t_rcvudata (3)\ \- \ 
receive unit data.
.HP
.BR t_rcvuderr (3)\ \- \ 
receive unit data error.
.HP
.BR t_rcvv (3)\ \- \ 
receive data and scatter.
.HP
.BR t_rcvvopt (3)\ \- \ 
receive data with options and scatter.
.HP
.BR t_rcvvudata (3)\ \- \ 
receive unit data and scatter.
.HP
.BR t_removeleaf (3)\ \- \ 
remove a point from a point-to-multipoint connection.
.HP
.BR t_snd (3)\ \- \ 
send transport data.
.HP
.BR t_snddis (3)\ \- \ 
send abortive disconnect.
.HP
.BR t_sndopt (3)\ \- \ 
send transport data with options.
.HP
.BR t_sndrel (3)\ \- \ 
send orderly release.
.HP
.BR t_sndreldata (3)\ \- \ 
send orderly release with data.
.HP
.BR t_sndudata (3)\ \- \ 
send unit data.
.HP
.BR t_sndv (3)\ \- \ 
gather and send data.
.HP
.BR t_sndvopt (3)\ \- \ 
gather and send data with options.
.HP
.BR t_sndvudata (3)\ \- \ 
gather and send unit data.
.HP
.BR t_strerror (3)\ \- \ 
describe an XTI/TLI library error.
.HP
.BR t_sync (3)\ \- \ 
synchronize XTI/TLI library with TLI provider.
.HP
.BR t_sysconf (3)\ \- \ 
configure XTI/TLI library parameters.
.HP
.BR t_unbind (3)\ \- \ 
unbind a transport endpoint.
.PD
.\"
.\".SS "Sockets Library"
.\".PP
.\".HP
.\".BR libsocket (3)\ \- \ 
.\".PD 0
.\".HP
.\".BR accept (3)\ \- \ 
.\".HP
.\".BR bind (3)\ \- \ 
.\".HP
.\".BR connect (3)\ \- \ 
.\".HP
.\".BR getpeername (3)\ \- \ 
.\".HP
.\".BR getsockname (3)\ \- \ 
.\".HP
.\".BR getsockopt (3)\ \- \ 
.\".HP
.\".BR listen (3)\ \- \ 
.\".HP
.\".BR recv (3)\ \- \ 
.\".HP
.\".BR recvfrom (3)\ \- \ 
.\".HP
.\".BR recvmsg (3)\ \- \ 
.\".HP
.\".BR send (3)\ \- \ 
.\".HP
.\".BR sendmsg (3)\ \- \ 
.\".HP
.\".BR sendto (3)\ \- \ 
.\".HP
.\".BR setsockopt (3)\ \- \ 
.\".HP
.\".BR shutdown (3)\ \- \ 
.\".HP
.\".BR socket (3)\ \- \ 
.\".HP
.\".BR socketpair (3)\ \- \ 
.\".HP
.\".BR sockets (3)\ \- \ 
.\".HP
.\".BR sockmod (4)\ \- \ 
.\".HP
.\".BR socksys (4)\ \- \ 
.\".HP
.\".BR protosw (5)\ \- \ 
.\".PD
.\"
.\".SS "NSL Library"
.\".PP
.\".HP
.\".BR libxnsl (3)\ \- \ 
.\".PD 0
.\".HP
.\".BR xnsl (3)\ \- \ 
.\".HP
.\".BR endnetconfig (3)\ \- \ 
.\".HP
.\".BR endnetpath (3)\ \- \ 
.\".HP
.\".BR freenetconfigent (3)\ \- \ 
.\".HP
.\".BR getnetconfig (3)\ \- \ 
.\".HP
.\".BR getnetconfigent (3)\ \- \ 
.\".HP
.\".BR getnetpath (3)\ \- \ 
.\".HP
.\".BR inetaddr (3)\ \- \ 
.\".HP
.\".BR nc_perror (3)\ \- \ 
.\".HP
.\".BR nc_sperror (3)\ \- \ 
.\".HP
.\".BR netconfig (3)\ \- \ 
.\".HP
.\".BR netdir (3)\ \- \ 
.\".HP
.\".BR netdir_free (3)\ \- \ 
.\".HP
.\".BR netdir_getbyaddr (3)\ \- \ 
.\".HP
.\".BR netdir_getbyname (3)\ \- \ 
.\".HP
.\".BR netdir_options (3)\ \- \ 
.\".HP
.\".BR netdir_perror (3)\ \- \ 
.\".HP
.\".BR netdir_sperror (3)\ \- \ 
.\".HP
.\".BR netpath (3)\ \- \ 
.\".HP
.\".BR setnetconfig (3)\ \- \ 
.\".HP
.\".BR setnetpath (3)\ \- \ 
.\".HP
.\".BR taddr2uaddr (3)\ \- \ 
.\".HP
.\".BR uaddr2taddr (3)\ \- \ 
.\".HP
.\".BR netconfig (5)\ \- \ 
.\".HP
.\".BR netdir (5)\ \- \ 
.\".HP
.\".BR test-xnsl (8)\ \- \ 
.\".HP
.\".BR test-xnsl_thr (8)\ \- \ 
.\".PD
.\"
.SS "TPI Interface"
.PP
.HP
.BR tpi (7)\ \- \ 
Transport Provider Interface
.PD 0
.HP
.BR T_ADDR_ACK (7)\ \- \ 
protocol address acknowledgement.
.HP
.BR T_ADDR_REQ (7)\ \- \ 
get protocol address request.
.HP
.BR T_BIND_ACK (7)\ \- \ 
bind protocol address acknowledgement.
.HP
.BR T_BIND_REQ (7)\ \- \ 
bind protocol address request.
.HP
.BR T_CAPABILITY_ACK (7)\ \- \ 
protocol capability acknowledgement.
.HP
.BR T_CAPABILITY_REQ (7)\ \- \ 
protocol capability acknowledgement.
.HP
.BR T_CONN_CON (7)\ \- \ 
connection confirm.
.HP
.BR T_CONN_IND (7)\ \- \ 
connect indication.
.HP
.BR T_CONN_REQ (7)\ \- \ 
connect request.
.HP
.BR T_CONN_RES (7)\ \- \ 
connection response.
.HP
.BR T_DATA_IND (7)\ \- \ 
data indication.
.HP
.BR T_DATA_REQ (7)\ \- \ 
data request.
.HP
.BR T_DISCON_IND (7)\ \- \ 
disconnect indication.
.HP
.BR T_DISCON_REQ (7)\ \- \ 
disconnect request.
.HP
.BR T_ERROR_ACK (7)\ \- \ 
error acknowledgement.
.HP
.BR T_EXDATA_IND (7)\ \- \ 
expedited data indication.
.HP
.BR T_EXDATA_REQ (7)\ \- \ 
expedited data request.
.HP
.BR T_INFO_ACK (7)\ \- \ 
protocol information acknowledgement.
.HP
.BR T_INFO_REQ (7)\ \- \ 
get transport protocol parameter sizes.
.HP
.BR T_OK_ACK (7)\ \- \ 
success acknowledgement.
.HP
.BR T_OPTDATA_IND (7)\ \- \ 
data indication with options.
.HP
.BR T_OPTDATA_REQ (7)\ \- \ 
data request with options.
.HP
.BR T_OPTMGMT_ACK (7)\ \- \ 
options management acknowledgement.
.HP
.BR T_OPTMGMT_REQ (7)\ \- \ 
options management.
.HP
.BR T_ORDREL_IND (7)\ \- \ 
orderly release indication.
.HP
.BR T_ORDREL_REQ (7)\ \- \ 
orderly release request.
.HP
.BR T_UDERROR_IND (7)\ \- \ 
unit data error indication.
.HP
.BR T_UNBIND_REQ (7)\ \- \ 
unbind protocol address request.
.HP
.BR T_UNITDATA_IND (7)\ \- \ 
unit data indication.
.HP
.BR T_UNITDATA_REQ (7)\ \- \ 
unit data request.
.PD
.\"
.SS "Test Programs"
.HP
.BR test-timod (8)\ \- \ 
a test suite executable for the
.BR timod (4)
.I STREAMS
module.
.PD 0
.HP
.BR test-tirdwr (8)\ \- \ 
a test suite executable for the
.BR tirdwr (4)
.I STREAMS
module.
.HP
.BR test-xnet (8)\ \- \ 
a test suite executable for the
.BR xnet (3)
.I STREAMS
library.
.HP
.BR test-xnet_thr (8)\ \- \ 
a test suite executable for the
.BR xnet (3)
.I STREAMS
library.
.PD
.PP
In addition to these manual pages, the
.B @PACKAGE_TITLE@
package installs an
.I Installation and Reference Manual
as well as a reprint of the
.I UNIX International
Transport Provider Interface specification.
.\"
.\"
.SH MODULES
.PP
.TP
.BR tirdwr (4)
a
.I STREAMS
XTI/TLI read write compatibility module.
.TP
.BR timod (4)
a
.I STREAMS
XTI/TLI compatibility interface.
.\"
.\"
.SH DEVICES
.PP
The
.B @PACKAGE_TITLE@
package does not provide any drivers of devices of its own.  Other packages
might define devices that autopush the
.BR timod (4)
or
.BR tirdwr (4)
modules.
.\"
.\"
.SH PREREQUISITES
.PP
The
.B @PACKAGE@
package requires the following packages to be installed:
.TP
.BR streams (5)
The
.B Linux Fast-STREAMS
base package.
.TP
.BR strcompat (5)
The
.B Linux Fast-STREAMS
compatiblity modules package.
.\"
.\"
.SH NOTICES
.PP
Although previous release of the
.B @PACKAGE@
package were tested with
.IR LiS ,
.I LiS
is now deprecated and this release
.RB ( @VERSION@ )
of
.B @PACKAGE@
has not been tested with
.IR LiS .

NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strxnet.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strxnet:	start and stop strxnet subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strxnet
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS XNET subsystem is configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strxnet
# Required-Start: streams $network
# Required-Stop: streams $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS XNET subsystem
# License: GPL
# Description:	This STREAMS XNET init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS XNET character
#	devices are present in the /dev directory and that the STREAMS XNET
#	subsystem is configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strxnet'
config="/etc/default/$name"
desc="the STREAMS XNET subsystem"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRXNET_MKNOD" ; then
    for STRXNET_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRXNET_MKNOD ] ; then
	    break
	else
	    STRXNET_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRXNET_PRELOAD"       ] || STRXNET_PRELOAD=""
[ -n "$STRXNET_DRIVERS"       ] || STRXNET_DRIVERS=""
[ -n "$STRXNET_MODULES"       ] || STRXNET_MODULES="streams-timod streams-tirdwr"
[ -n "$STRXNET_MAKEDEVICES"   ] || STRXNET_MAKEDEVICES="yes"
[ -n "$STRXNET_REMOVEDEVICES" ] || STRXNET_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRXNET_MKNOD" ] && STRXNET_MAKEDEVICES="no"
[ -z "$STRXNET_MKNOD" ] && STRXNET_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRXNET_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRXNET_MKNOD" -a ":$STRXNET_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS XNET devices: "
	$STRXNET_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRXNET_MKNOD" -a ":$STRXNET_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS XNET devices: "
	$STRXNET_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRXNET_PRELOAD $STRXNET_DRIVERS $STRXNET_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.10  2008-10-21 22:42:22  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.9  2008-04-28 18:38:42  brian
# - header updates for release
#
# Revision 0.9.2.8  2007/08/14 04:01:30  brian
# - GPLv3 header update
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
